docs/DeveloperGuide.md

```diff
@@ -1,34 +1,247 @@
+
+`entry-name, start-time, end-time, duration, tags`
+
+All tags are saved in the same cell, separated by a white space; this design decision was taken to make sure that each entry occupies the same number of cells regardless of each entry’s number of tags. The tags are then separately parsed when the data is loaded.
+
+Each Storage objects contains the path to the data file (`Storage.dataFilePath`), the File object representing the data file (`Storage.dataFile`), and an activityList populated with the data from the data file (`Storage.activityList`). Storage optionally supports multiple data files at the same time, allowing implementation of features like multiple sessions and multiple user profiles. 
+
+Storage provides the following functions:
+- Constructing a Storage object via `Storage(String dataFilePath)`, which takes in the path to the desired data file (or the path where the user wants to create the data file) as a String object.
+- Creating a data file via `createDataFile`.
+- Writing to a data file via `writeToFile`. This function takes a single string as parameter and writes it to the data file. It is recommended to only pass single-line strings to keep the file nicely formatted. 
+Loading a pre-existing data file via `loadFile`. If a data file already exists for the provided data file path, the function will return `true`; if the specified data file did not previously exist, this function will call the `createDataFile` method and returns `false`. The return value is useful so that the application knows whether or not this is the first session with a specific data file or if data already exists.
+- Creating an ActivityList via `createActivityList`. This function calls `loadFile()` to check whether the file already existed or not; if the data file previously existed, it will construct an ActivityList object by passing the data from the data file to it, and return this populated ActivityList object; if the data file did not previously exist, it will return an empty activityList object.
+
+### 3.3 Storage handler
+The StorageHandler class functions as a support to the main Storage class, allowing the Jikan application to manipulate the stored data file. Its main provided functions are:
+- Removing an entry from the data file via `removeLine`. This function takes in the number of the line to remove.
+- Replacing an entry in the data file via `replaceLine`. This function takes in the number of the line to replace, along with the String object that needs to be written to the data file in place of the replaced line.
+
+### 3.4 Edit feature
+The edit feature allows the user to make changes to activities that have been saved in the activity list. This is to allow the user to rectify any mistakes that may have been made during the initial recording of the activity. 
+
+#### 3.4.1 Current Implementation
+The following sequence diagram shows how the edit feature works.
+![image_info](./pictures/EditSequenceDiagram.png)
```

Comment:
Similiar to the comment that I gave above, the lines in the diagram are difficult to be seen. Perhaps it would be better to change the colour of the lines in the diagram so that the diagram looks more visible for the reader?